v2/internal/genericarmclient/generic_client.go

```diff
@@ -115,16 +118,26 @@ func (client *GenericClient) BeginCreateOrUpdateByID(ctx context.Context, resour
 
 func (client *GenericClient) createOrUpdateByID(
 	ctx context.Context,
-	resourceID string,
+	resourceID,
```

Comment:
Same here - cascading several parameter types together invites misinterpretation. Be explicit.